From ac39d79c38ffc819dfff5f9e82e175f4d8a72873 Mon Sep 17 00:00:00 2001
From: Guo Mengqi <guomengqi3@huawei.com>
Date: Tue, 2 Aug 2022 11:10:31 +0800
Subject: [PATCH 4361/4585] [Huawei] mm: sharepool: proc_sp_group_state bugfix

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5K3MH

--------------------------------

After refactoring, cat /proc/pid_xx/sp_group will cause kernel panic.
Fix this error.

Signed-off-by: Guo Mengqi <guomengqi3@huawei.com>
---
 mm/share_pool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index 73a80548cf02..a7e91867b1a8 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -3995,7 +3995,7 @@ int proc_sp_group_state(struct seq_file *m, struct pid_namespace *ns,
 	seq_printf(m, "%-8s %-9s %-9s %-9s %-4s\n",
 			"Group_ID", "SP_ALLOC", "SP_K2U", "SP_RES", "PROT");
 
-	list_for_each_entry(spg_node, &master->node_list, proc_node) {
+	list_for_each_entry(spg_node, &master->node_list, group_node) {
 		seq_printf(m, "%-8d %-9ld %-9ld %-9ld ",
 				spg_node->spg->id,
 				get_spg_proc_alloc(spg_node),
-- 
2.17.1

